src/Listener/ODM/MediaEventSubscriber.php

```diff
@@ -31,18 +33,26 @@ public function getSubscribedEvents()
         ];
     }
 
-    protected function recomputeSingleEntityChangeSet(EventArgs $args): void
+    protected function recomputeSingleEntityChangeSet(LifecycleEventArgs $args): void
     {
+        \assert($args instanceof ODMLifecycleEventArgs);
+
         $em = $args->getDocumentManager();
 
         $em->getUnitOfWork()->recomputeSingleDocumentChangeSet(
-            $em->getClassMetadata(\get_class($args->getDocument())),
-            $args->getDocument()
+            $em->getClassMetadata(\get_class($args->getObject())),
+            $args->getObject()
         );
     }
 
-    protected function getMedia(EventArgs $args)
+    protected function getMedia(LifecycleEventArgs $args)
     {
-        return $args->getDocument();
+        $media = $args->getObject();
+
+        if (!$media instanceof MediaInterface) {
+            throw new \RuntimeException('There is no media on the persistence event.');
```

Comment:
I'd say `\InvalidArgumentException` or `\LogicException` are more accurate in this case.

Reply:
Fixed